src/components/DayStreak.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from "react";
+import { Text } from "react-native";
+import styles from "../../styles";
+import AsyncStorage from '@react-native-community/async-storage';
+
+/*
+ * Day streak. Display how many consecutive days the user has
+ * meditated for.
+ */
+const DayStreak = ({ navigation }: any) => {
+  const [streak, setStreak] = useState("");
+
+  /*
+   * Get the current count and the last day the user meditated.
+   */
+  const getStreakData = async () => {
+    // Worry what happens here if no key is found, however testing
+    // suggests this is fine.
+    try {
+      // Update last streak count and last day meditated
+      return {
+        count: Number(await AsyncStorage.getItem('@streak_Count')),
+        lastDay: new Date(Number(await AsyncStorage.getItem('@last_Day_Year')),
+                          Number(await AsyncStorage.getItem('@last_Day_Month')),
+                          Number(await AsyncStorage.getItem('@last_Day_Date')))
+      }
+    } catch (e) {
+      // Error getting data...
+    }
+    return {
+      // We hit an error, just return null
+      count: null,
+      lastDay: null
+    }
+  }
+
+  /*
+   * Save the current count and today's date meditated.
+   */
+  const updateStreakData = async (count: any,
+                                  currentDay: Date) => {
+    try {
+      // Update last streak count and last day meditated
+      // Async Storage can only store strings in key value pairs,
+      // so convert the count to a string.
+      await AsyncStorage.setItem('@streak_Count', String(count));
+      await AsyncStorage.setItem('@last_Day_Date', String(currentDay.getDate()));
+      await AsyncStorage.setItem('@last_Day_Month', String(currentDay.getMonth()));
+      await AsyncStorage.setItem('@last_Day_Year', String(currentDay.getFullYear()));
+    } catch (e) {
+      // Error saving data, fail quietly...
+    }
+  }
+
+  /*
+   * Calculate the current streak. That is, the number of consecutive
+   * days the user has meditated for.
+   */
+  const getDayStreak = async () => {
+    try {
+      // Get the stored data.
+      var StreakData = await getStreakData();
+      var count =  StreakData.count;
+      var lastDay = StreakData.lastDay;
```

Comment:
You can use destructuring here: `const {count, lastDay} = streakData`

Reply:
ah hit issue here where I try and change count later on in the code...

Reply:
Is there a way around this?

Reply:
It will need to be a let

Reply:
ah ta